data/helpers.d/psql

```diff
@@ -1,148 +1,273 @@
+# usage: ynh_psql_execute_as_root --sql=sql [--database=database]
+# | arg: -s, --sql - the SQL command to execute
+# | arg: -d, --database - the database to connect to
+ynh_psql_execute_as_root() {
+	# Declare an array to define the options of this helper.
+	local legacy_args=sd
+	declare -Ar args_array=([s]=sql= [d]=database=)
+	local sql
+	local database
+	# Manage arguments with getopts
+	ynh_handle_getopts_args "$@"
+	database="${database:-}"
+
+	ynh_psql_connect_as --user="postgres" --password="$(sudo cat $PSQL_ROOT_PWD_FILE)" \
+		--database="$database" <<<"$sql"
 }
 
 # Execute a command from a file as root user
 #
-# usage: ynh_psql_execute_file_as_root file [db]
-# | arg: file - the file containing SQL commands
-# | arg: db - the database to connect to
+# usage: ynh_psql_execute_file_as_root --file=file [--database=database]
+# | arg: -f, --file - the file containing SQL commands
+# | arg: -d, --database - the database to connect to
 ynh_psql_execute_file_as_root() {
-	local file="$1"
-	local db="$2"
-	sudo --login --user=postgres psql "$db" < "$file"
-}
+	# Declare an array to define the options of this helper.
+	local legacy_args=fd
+	declare -Ar args_array=([f]=file= [d]=database=)
+	local file
+	local database
+	# Manage arguments with getopts
+	ynh_handle_getopts_args "$@"
+	database="${database:-}"
 
-# Create a database, an user and its password. Then store the password in the app's config
-#
-# After executing this helper, the password of the created database will be available in $db_pwd
-# It will also be stored as "psqlpwd" into the app settings.
-#
-# usage: ynh_psql_setup_db user name [pwd]
-# | arg: user - Owner of the database
-# | arg: name - Name of the database
-# | arg: pwd - Password of the database. If not given, a password will be generated
-ynh_psql_setup_db () {
-	local db_user="$1"
-	local db_name="$2"
-	local new_db_pwd=$(ynh_string_random)	# Generate a random password
-	# If $3 is not given, use new_db_pwd instead for db_pwd.
-	local db_pwd="${3:-$new_db_pwd}"
-	ynh_psql_create_db "$db_name" "$db_user" "$db_pwd"	# Create the database
-	ynh_app_setting_set "$app" psqlpwd "$db_pwd"	# Store the password in the app's config
+	ynh_psql_connect_as --user="postgres" --password="$(sudo cat $PSQL_ROOT_PWD_FILE)" \
+		--database="$database" <"$file"
 }
 
-# Create a database and grant privilegies to a user
+# Create a database and grant optionnaly privilegies to a user
 #
-# usage: ynh_psql_create_db db [user [pwd]]
+# [internal]
+#
+# usage: ynh_psql_create_db db [user]
 # | arg: db - the database name to create
 # | arg: user - the user to grant privilegies
-# | arg: pwd  - the user password
 ynh_psql_create_db() {
-	local db="$1"
-	local user="$2"
-	local pwd="$3"
-	ynh_psql_create_user "$user" "$pwd"
-	sudo --login --user=postgres createdb --owner="$user" "$db"
+	local db=$1
+	local user=${2:-}
+
+    local sql="CREATE DATABASE ${db};"
+
+    # grant all privilegies to user
+	if [ $# -gt 1 ]; then
```

Comment:
Hm I guess this is meant to check if `$user` is not empty ? Why not use `[[ -n "$user" ]]` to be more explicit ? :sweat_smile:

Reply:
Without [[ ]] !
Otherwise, it would work as well.